File: problems/3270-find-the-key-of-the-numbers/solution.java
class Solution {
    public int generateKey(int num1, int num2, int num3) {
        String s1 = String.format("%04d", num1);
        String s2 = String.format("%04d", num2);
        String s3 = String.format("%04d", num3);
        String key = "";
        for (int i = 0; i < 4; i++) {
            key += Math.min(s1.charAt(i), Math.min(s2.charAt(i), s3.charAt(i)));
        }
        return Integer.parseInt(key);
    }
}

File: problems/3270-find-the-key-of-the-numbers/solution.c
int generateKey(int num1, int num2, int num3) {
    char s1[5], s2[5], s3[5];
    sprintf(s1, "%04d", num1);
    sprintf(s2, "%04d", num2);
    sprintf(s3, "%04d", num3);
    char key[5] = "";
    for (int i = 0; i < 4; i++) {
        key[i] = min(min(s1[i], s2[i]), s3[i]);
    }
    key[4] = '\0';
    return atoi(key);
}